internal/search/run/expression_job_test.go

```diff
@@ -0,0 +1,166 @@
+package run
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/search"
+	"github.com/sourcegraph/sourcegraph/internal/search/result"
+	"github.com/sourcegraph/sourcegraph/internal/search/streaming"
+)
+
+func TestAndJob(t *testing.T) {
+	t.Run("NewAndJob", func(t *testing.T) {
+		t.Run("no children is simplified", func(t *testing.T) {
+			require.Equal(t, NewNoopJob(), NewAndJob())
+		})
+		t.Run("one child is simplified", func(t *testing.T) {
+			j := NewMockJob()
+			require.Equal(t, j, NewAndJob(j))
+		})
+	})
+
+	t.Run("result returned from all subexpressions is streamed", func(t *testing.T) {
+		res := &result.RepoMatch{Name: "test", ID: 1}
+		mj := NewMockJob()
+		mj.RunFunc.SetDefaultHook(func(_ context.Context, _ database.DB, s streaming.Sender) (*search.Alert, error) {
+			s.Send(streaming.SearchEvent{Results: []result.Match{res}})
+			time.Sleep(10 * time.Millisecond)
+			return nil, nil
+		})
+
+		for i := 2; i < 5; i++ {
+			t.Run(fmt.Sprintf("%d subexpressions", i), func(t *testing.T) {
+				var subexpressions []Job
+				for j := 0; j < i; j++ {
+					subexpressions = append(subexpressions, mj)
+				}
+
+				j := NewAndJob(subexpressions...)
+
+				start := time.Now()
+				var eventTime time.Time
+				s := streaming.StreamFunc(func(e streaming.SearchEvent) {
```

Comment:
```suggestion
				s := streaming.StreamFunc(func(_ streaming.SearchEvent) {
```

(nit to assert I actually read through the code 😝 )